src/graphs/Dijkstra.re

```diff
@@ -0,0 +1,151 @@
+type node = {
+    id: string,
+    neighbours: list(string),
+};
+
+type directedGraph = list(node);
+
+type edge = {
+    tail: string,
+    head: string,
+    weight: int,
+};
+
+type resultType = {
+    dist: Hashtbl.t(string, int),
+    prev: Hashtbl.t(string, option(string))
+};
+
+exception Not_found(string);
+
+let infinity = Pervasives.max_int;
+
+let parseAdjList = adj_list => {
```

Comment:
I think we should construct the adj_table from the edges instead. We don't need to specify the adj_list at all!